### PR TITLE
feat: drop Python 2

### PR DESCRIPTION
**Pull request by henryiii**
With Uproot and Awkward dropping Python 2, I think it's time to do it here too!

### CODE REVIEW

File: src/particle/particle/particle.py [hunk 1]
```diff
@@ -1124,9 +1121,6 @@ def finditer(
         See also ``findall``, which returns the same thing, but as a list.
         """
 
-        # Note that particle can be called by position to keep compatibility with Python 2, but that behavior should
-        # not be used and will be removed when support for Python 2.7 is dropped.
-
```
Comment (henryiii) [inline src/particle/particle/particle.py:1129]: I've implemented this now, as promised.

**Review by eduardo-rodrigues — APPROVED** [review]
This all looks good to me. Thank you 👍!

File: src/particle/particle/particle.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply src/particle/particle/particle.py:1129]: To what are you refering to?

Comment (henryiii) [reply src/particle/particle/particle.py:1129]: `finditer(function, True)` is no longer allowed, now it must be `finditer(function, particle=True)`. Though I missed `findall` just below. Will fix.

**Pull request merged**